lutris/gui/views/base.py

```diff
@@ -28,35 +28,57 @@ def popup_contextual_menu(self, view, event):
-        view.current_path = view.get_path_at_pos(event.x, event.y)
-        if view.current_path:
-            view.select()
-            model = self.get_model()
-            _iter = model.get_iter(view.current_path[0])
-            if not _iter:
-                return
-            col_id = str(model.get_value(_iter, COL_ID))
-
-            game_actions = self.get_game_actions(col_id)
+        game_ids = []
+        if self.view_type == "grid":
+            selection = self.get_selected_items()
+        else:
+            selection = self.get_selection().get_selected_rows()
+            selection = selection[1]
+        if len(selection) < 2:
+            view.current_path = view.get_path_at_pos(event.x, event.y)
+            if view.current_path:
+                view.select()
+                model = self.get_model()
+                _iter = model.get_iter(view.current_path[0])
+                if not _iter:
+                    return
+                col_id = str(model.get_value(_iter, COL_ID))
+                game_ids.append(col_id)
+                game_actions = self.get_game_actions(col_id)
+                if game_actions:
+                    contextual_menu = ContextualMenu(game_actions.get_game_actions())
+                    contextual_menu.popup(event, game_actions)
+        else:
+            for path in selection:
+                iterator = view.get_model().get_iter(path)
+                game_id = view.get_model().get_value(iterator, COL_ID)
+                game_ids.append(game_id)
+            game_actions = self.get_game_actions(game_ids)
             if game_actions:
-                contextual_menu = ContextualMenu(game_actions.get_game_actions())
+                contextual_menu = ContextualMenu(game_actions.get_multiple_game_actions())
                 contextual_menu.popup(event, game_actions)
 
-    def get_game_actions(self, game_id):
-        if self.service:
-            db_game = get_game_for_service(self.service.id, game_id)
 
-            if db_game:
-                game = self.get_game_by_id(db_game["id"])
+    def get_game_actions(self, game_ids):
+        games = []
+        for game_id in game_ids:
+            if self.service:
+                db_game = get_game_for_service(self.service.id, game_id)
+
+                if db_game:
+                    game = self.get_game_by_id(db_game["id"])
+                else:
+                    db_game = ServiceGameCollection.get_game(self.service.id, game_id)
+                    game = Game.create_empty_service_game(db_game, self.service)
+            elif game_id:
+                game = self.get_game_by_id(game_id)
             else:
-                db_game = ServiceGameCollection.get_game(self.service.id, game_id)
-                game = Game.create_empty_service_game(db_game, self.service)
-        elif game_id:
-            game = self.get_game_by_id(game_id)
+                return None
+            games.append(game)
+        if len(games) == 1:
```

Comment:
This len check should live in get_game_actions() so it's harder to forget to do it.